aep/general/0008/aep.md

```diff
@@ -1,5 +1,259 @@
+rationale instead of other sections, to ensure the rest of the document acts as
+an easily actionable reference.
+
+## History
+
+The "history" section is optional, and documents events and context around a
+significant edit to an AEP. For example, explanation of rewrite would be
+included in this section
+
+While the changelog is a dotted list of one-line summaries of changes to an AEP,
+the history section should elaborate on significant events in a descriptive
+format.
+
+The section **must not** be used to exhaustively enumerate all changes. This
+is what the changelog provides.
+
+## Further reading
+
+A bulleted list of (usually) other AEPs, in the following format:
+
+- [AEP-1](./0001.md): AEP purpose and guidelines
+
+## Changelog
+
+A bulleted list of changes in reverse chronological order, using the following
+format:
+
+- **2020-02-18**: Specified ordering.
+- **2019-07-01**: Added a subsection clarifying XYZ.
+```
+
+AEPs **should** attempt to follow this overall format if possible, but AEPs
+**may** deviate from it if necessary (in particular, if the AEP would be more
+difficult to understand, even for a reader already accustomed to reading AEPs in
+the usual format).
+
+**Note:** Except for the title, AEPs **must** only use the second heading level
+(`##`) and above. AEPs **should** only use the second and third heading levels
+(`##`, `###`).
+
+### Requirement keywords
+
+AEPs **should** use the following requirement level keywords: "MUST", "MUST
+NOT", "SHOULD", "SHOULD NOT", and "MAY", which are to be interpreted as
+described in [RFC 2119][].
+
+When using these terms in AEPs, they **must** be lower-case and **bold**. These
+terms **should not** be used in other ways.
+
+If "SHOULD" or "SHOULD NOT" are used, they **must** include valid examples of
+where other concerns may override the guidance.
+
+**Important:** If rationale is used, it exists to provide background and a more
+complete understanding, but **must not** contain guidance (and RFC-2119 terms
+**must not** be used).
+
+### Code examples
+
+API design examples in AEPs **should** be presented in both [OpenAPI][] and
+[protocol buffers][]. Examples
+**should** cover only enough syntax to explain the concept. When using RPCs in
+examples, a `library.api.http` annotation **should** be included.
+
+AEPs should use snake-case naming for parameters and properties (fields), for
+consistency across OpenAPI and protocol buffers, but other casing conventions
+may be used as long as they are applied uniformly.
+
+### Referencing AEPs
+
+When AEPs reference other AEPs, the prosaic text **must** use the format
+`AEP-XXXX` without zero-padding (e.g., `AEP-8`, not `AEP-0008`), and **must**
+link to the relevant AEP. AEP links **may** point to a particular section of the
+AEP if appropriate.
+
+**Important:** AEP links **must** use the relative path to the file in the
+repository (such as `./0008.md` for core AEPs, or `../0008.md` for AEPs in a
```

Comment:
Same question about slugs, though I'm not sure how the site generator would handle slug-based links rather than filename-based links.  But filename-based links won't get updated if we go with the plan of moving slugs to new files for AEP versioning.

(But maybe this is good, if we want e.g. two old versions of AEPs to maintain a link between them, rather than e.g. the old DELETE pointing to the new SOFT-DELETE.)

Maybe this is a bigger question and shouldn't block this PR.

@toumorokoshi FYI